packages/loopring_v3/contracts/amm/libamm/AmmExitRequest.sol

```diff
@@ -24,97 +24,71 @@ library AmmExitRequest
     using SafeCast          for uint;
     using SignatureUtil     for bytes32;
 
-    bytes32 constant public WITHDRAW_TYPEHASH = keccak256(
-        "Withdraw(address owner,uint256[] amounts,uint256 validUntil,uint256 nonce)"
-    );
-
+    // TODO:fix this string
     bytes32 constant public POOLEXIT_TYPEHASH = keccak256(
-        "PoolExit(address owner,bool toLayer2,uint256 poolAmountIn,uint256[] minAmountsOut,uint32[] storageIDs,uint256 validUntil)"
+        "PoolExit(address owner,uint96 burnAmount,bool burnFromLayer2,uint32 burnStorageID,uint96[] exitMinAmounts,bool exitToLayer2,uint256 validUntil)"
     );
 
-    event Withdrawal(address owner, uint[] amountOuts);
     event PoolExitRequested(AmmData.PoolExit exit);
-    event UnlockScheduled(address owner, uint timestamp);
-
-    function unlock(AmmData.State storage S)
-        internal
-        returns (uint lockedUntil)
-    {
-        require(S.lockedUntil[msg.sender] == 0, "UNLOCKED_ALREADY");
-        require(S.lockedSince[msg.sender] >= block.timestamp, "LOCK_PENDING");
-
-        lockedUntil = block.timestamp + AmmData.LOCK_DELAY();
-
-        S.lockedSince[msg.sender] = 0;
-        S.lockedUntil[msg.sender] = lockedUntil;
 
-        emit UnlockScheduled(msg.sender, lockedUntil);
-    }
-
-    function withdrawFromPool(
+    function exitPool(
         AmmData.State storage S,
-        uint[]       calldata amounts,
-        bytes        calldata signature, // signature from Exchange operator
-        uint                  validUntil
+        AmmData.Direction     direction,
+        uint96                burnAmount,
+        uint32                burnStorageID,
+        uint96[]     calldata exitMinAmounts
         )
         public
-        returns (uint[] memory amountOuts)
     {
-        uint size = S.tokens.length;
-        require(amounts.length == size + 1, "INVALID_DATA");
+        uint size = S.tokens.length - 1;
+        require(exitMinAmounts.length == size, "INVALID_PARAM_SIZE");
+        require(burnAmount > 0, "INVALID_BURN_AMOUNT");
 
-        _proxcessExchangeWithdrawalApprovedWithdrawals(S);
-
-        bool approvedByOperator = _checkOperatorApproval(
-            S, amounts, signature, validUntil
+        bool fromLayer1 = (
+            direction == AmmData.Direction.L1_TO_L1 ||
+            direction == AmmData.Direction.L1_TO_L2
         );
 
-        amountOuts = new uint[](size + 1);
-        amountOuts[0] = _withdrawToken(
-            S, address(this), amounts[0], approvedByOperator
-        );
+        uint32 lockIdx;
+        if (fromLayer1) {
+            require(burnStorageID == 0, "INVALID_STORAGE_ID");
+            require(S.exitLockIdx[msg.sender] == 0, "ONLY_ONE_LAYER1_EXIT_PER_USER_ALLOWED");
```

Comment:
Why is this done?

Reply:
I think exits can be simplified while keeping a similar mechanism (exits can be done out-of-order):
- Just keep a counter to limit the number of exits (increment when creating the request, decrement when processing it)
- Store the amount like `S.lockedToBurn[exit.owner] = exit.burnAmount`, if `S.lockedToBurn[exit.owner]  > 0` user is already exiting.

Reply:
I want to make exits similar to layer AMMs, they process exits by order.